### DISCUSSION TITLE
RFC 16 - Better Tags

### DISCUSSION THREAD

**Discussion by SeriousBug** [dendronhq/dendron#885] [RFCs/Ideas]
This is the discussion for the [RFC 16 - Better Tags](https://wiki.dendron.so/notes/NT1cFX6DRkTnzcWwduj2I.html).

**Comment by SR--** [comment]
Displaying and even manipulating tags within the graph could be useful.

**Reply by kevinslin** [reply]
Something that'll be good to mention is colors for tags. Since we have full control of the HTML, we could generate a color for each tag. This might be customizable in the tag note via frontmatter (eg. tagColor)

**Comment by boschet** [comment]
I think that separating the tags from the graph structure would be very useful as the different information domains are separated. As of now I'd prefer frontmatter tags as it does not interfere with the actual text. Although I can see a use case for inline tags:
When I summarize my lecture notes in dendron and then later prepare for my exams I need to mark specific formulae I need for the exam. Usually that is a lot less than the whole note, but can't be atomized in smaller notes. 
There is a tradeoff between the readability of the markdown and further structuring the notes.

Another nice feature would be a conversion tool that converts all the existing tags (I think of a tag branch in the graph) to whatever kind of tagging system is implemented.

**Comment by SR--** [comment]
What about colours for individual/specific tags or groups of tags?

**Reply by SeriousBug** [reply]
I was thinking of customizable colors for individual tags. What did you have in mind for groups of tags?

**Reply by SR--** [reply]
Well, I might have _admin_ tags for myself or collaborators (#urgent, #dodgy, etc.), _content_ tags about research (#ethics, #math, etc.), _workflow_ tags for the team (#john, #susan, etc.) and it might be useful to give these groups/types group colours...

**Reply by SeriousBug** [reply]
Do you think it would be a good workflow to individually set colors for these tags? E.g. for each tag note "tags.ethics", "tags.math" you can set the same color in the frontmatter.

Otherwise, I'd be happy to hear any ideas you have about how you could group these tags. Maybe something like "#research.ethics" and "#research.math"?

**Reply by SR--** [reply]
Yes, if tags are nested via the typical . system, it should be easy to give them hierarchical colours with domain colours propagating downwards by default. It would become especially powerful if lookup/renaming/refactoring commands could work on tags, images, etc. as well as on the file names. Once you have lots of tags with many settings it will be important to be able to list/unify/change them in bulk.

**Reply by SeriousBug** [reply]
Definitely agreed. Also; lookup, rename, and refactor should already work for tags.

**Reply by SR--** [reply]
Sounds good. Lookup, rename, refactor should be able to control tag-specific properties though somehow, such as colour.

**Comment by SR--** [comment]
Maybe options for the visibility of all and/or some tags or groups of tags in preview/publishing/graph?

**Reply by SeriousBug** [reply]
Filtering the graph by tags is definitely a feature I think would be useful, it's already mentioned in the RFC. I'm not sure what you mean by publishing though, do you mean hiding tags in published website so you can't see the tag? Or hiding notes in the published website that are tagged with a specific tag?

**Reply by SR--** [reply]
I meant hiding tags so that they or some of them are not visible. But hiding by tag might also be useful as you suggest.

**Reply by SeriousBug** [reply]
That's a good idea, tags should follow existing Dendron functionality to hide specific tags (for example, setting `published: false`). Once a tag is hidden, it should not only hide the tag page but also hide the tag from being displayed on rendered pages.

**Reply by SR--** [reply]
Sounds good.

**Comment by SR--** [comment]
Tag tree view? Currently available as the file/stub tree view but would benefit from some separation or tag-related control/filtering?

**Reply by SeriousBug** [reply]
All tags will go under "tags" in the tree view and backlinks, which will group them together. Perhaps we can come up with some way to emphasize them better, like putting a special symbol next to the tags entry or putting at the top.

**Reply by SR--** [reply]
Hm... It would be good to be able to intersect tags and file names in the tree. To only show the tree for tags #blah or to only show the tags for domain "math.something". Or to only show tags or only show note names. What do you think?

**Reply by SeriousBug** [reply]
The tags are internally just notes, so they would show up on the treeview like:
```
my-notes
other-notes
tags
├── admin
│   ├── urgent
│   └── dodgy
└── research
    ├── ethics
    └── math
```
I'm not fully sure if something entirely separate is needed, or if this is already a good way to display tags. Especially if we put the tags always at the top or always at the bottom (so it's easy to find them at a glance), you can then just look at tags, or just look at tags under a certain hierarchy (for example, just ones under "tags.admin" by expanding or narrowing the tree view.

**Reply by SR--** [reply]
I agree. Hard to know without trying. I was thinking about using this for knowledge discovery, browsing and narrowing down the tree to drill into its portions...
